metrics/codahale.go

```diff
@@ -179,13 +177,24 @@ func (c *CodaHale) MeasureResponse(code int, method string, routeId string, star
 
 func (c *CodaHale) MeasureServe(routeId, host, method string, code int, start time.Time) {
 	method = measuredMethod(method)
+	keyServeRoute := fmt.Sprintf("serveroute.%s", routeId)
+	keyServeHost := fmt.Sprintf("servehost.%s", hostForKey(host))
+
+	if c.options.EnableServeMethodMetric {
+		keyServeHost += fmt.Sprintf(".%s", method)
+		keyServeRoute += fmt.Sprintf(".%s", method)
+	}
+	if c.options.EnableServeStatusCodeMetric {
+		keyServeHost += fmt.Sprintf(".%d", code)
+		keyServeRoute += fmt.Sprintf(".%d", code)
+	}
```

Comment:
This will be executed even when `EnableServeRouteMetrics` or/and `EnableServeHostMetrics` are disabled.

Reply:
The same is for prometheus

Reply:
yeah, although I know we might have a small performance/memory penalty for that, I thought the code would be more "direct". I was undecided, thanks for pointing it out, I just pushed a fix. It's hard to test, thought.

Reply:
as there are some frequent repeated string operations here, I would consider a microoptimization like this:

```
var keyServeRoute, keyServeHost string
hfk := hostForKey(host)
switch {
case c.options.EnableServeMethodMetric && c.options.EnableServeStatusCodeMetric:
    keyServeHost = fmt.Sprintf("servehost.%s.%s.%d", hfk, method, code)
    keyServeRoute = fmt.Sprintf("serveroute.%s.%s.%d", hfk, method, code)
case c.options.EnableServeMethodMetric:
    keyServeHost = fmt.Sprintf("servehost.%s.%s", hfk, method)
    keyServeRoute = fmt.Sprintf("serveroute.%s.%s", hfk, method)
case c.options.EnableServeStatusCodeMetric:
    keyServeHost = fmt.Sprintf("servehost.%s.%d", hfk, code)
    keyServeRoute = fmt.Sprintf("serveroute.%s.%d", hfk, code)
default:
    keyServeHost = fmt.Sprintf("servehost.%s", hfk)
    keyServeRoute = fmt.Sprintf("serveroute.%s", hfk)
}
```

though it's true that the current approach might be easier to read

Reply:
I like this idea, @aryszka, I just had to change it slightly to include the routes ones:
```diff
- keyServeRoute = fmt.Sprintf("serveroute.%s.%s.%d", hfk, method, code)
+ keyServeRoute = fmt.Sprintf("serveroute.%s.%s.%d", routeId, method, code)
```